ciphey/basemods/Crackers/affine.py

```diff
@@ -0,0 +1,105 @@
+# community
+# by https://github.com/Ozzyz
+
+
+from distutils import util
+from typing import Optional, Dict, Union, Set, List
+from loguru import logger
+
+import ciphey
+from ciphey.iface import ParamSpec, Cracker, CrackResult, CrackInfo, T, registry, Config
+from ciphey.common import fix_case
+
+import re
+from math import gcd
+
+
+@registry.register
+class Affine(Cracker[str]):
+    def getInfo(self, ctext: str) -> CrackInfo:
+        return CrackInfo(
+            success_likelihood=0.1, success_runtime=1e-5, failure_runtime=1e-5,
+        )
+
+    @staticmethod
+    def getTarget() -> str:
+        return "Affine"
+
+    def attemptCrack(self, ctext: str) -> List[CrackResult]:
+        """
+        Brute forces all the possible combinations of a and b to attempt to crack the cipher.
+        Each character in the Affine Cipher is encoded with the rule E(x) = (ax + b) mod m
+        m is the size of the alfabet, while a and b are the keys in the cipher. a must be coprime to b.
+        The Caesar cipher is a specific case of the Affine Cipher, with a=1 and b being the shift of the cipher.
+        Decrypton is performed by D(x) = a_inv (x - b) mod m where a_inv is the modular multiplicative inverse of a mod m. 
+        """
+        logger.trace("Attempting Affine brute force.")
+        variants = []
+        candidates = []
+        result = []
```

Comment:
You can use unpacking here:

```
variants, candidates, result = [], [], []
```

Reply:
Yeah, I will do a general cleanup as soon as I have time, and these sort of things are part of that. Two of these aren't even used : )